squad/core/models.py

```diff
@@ -940,6 +940,8 @@ class ProjectStatus(models.Model, TestSummaryBase):
     last notification.
     """
     build = models.OneToOneField('Build', related_name='status', on_delete=models.CASCADE)
+    baseline = models.OneToOneField('Build', related_name='next_status', on_delete=models.SET_NULL, null=True, blank=True)
```

Comment:
IMHO, I think `next_build` field is redundant:

For example, let there be builds b1, b2, b3 and we want to lay them out in a way that exists today (a build baseline is always the most recent finished one):
* b1.baseline = None
* b2.baseline = b1, this will make b1.next_status = b2
* b3.baseline = b2, this will make b2.next_status = b3
* if b2 gets removed, b3.baseline = b1, making b1.next_status = b3

Does that make sense or am I missing something here?

Also, can two builds have a common baseline?

Reply:
If b2 gets deleted, ProjectStatus in b3 might have wrong numbers (as baseline changed). So my idea is to pin the baseline so we don't get this issue. 

Yes, you can have 2 builds with common baseline
 * build1 - finished
 * build2 - not finished (baseline build1)
 * build3 - not finished (baseline build1)

If build3 finishes before build2 (can happen) than without pinning it's baseline we might have numbers wrong, same as with deleting use case.